setuptools/command/editable_wheel.py

```diff
@@ -770,7 +780,9 @@ def _find_spec(cls, fullname, candidate_path):
         init = candidate_path / "__init__.py"
         candidates = (candidate_path.with_suffix(x) for x in module_suffixes())
         for candidate in chain([init], candidates):
-            if candidate.exists():
+            if candidate.is_file() and (
+                _relax_case() or _in_dir_case_sensitive(candidate, candidate.parent)
+            ):
```

Comment:
I wonder if it would work if we the finder just handle the top level names and rely on the existing machinery for the nested modules/packages... I have the impression that `importlib.machinery.FileFinder` would first try to import the parent package and then use the parent package `__path__` to resolve the nested path...

Not sure though, I have not tested. If that works, we could skip the checks all together and possibly simplify the template, right?

Reply:
I think you're right. I tried this a bit and got stuck with failing namespace packages, but I will try this route again.

Reply:
So apparently namespace packages behave differently and the import machine does not try to find nested packages based on the `__path__` property of the parent package...

Maybe in the future it, I can try to create a reproducer and bring it up in an issue in the python/cpython repository... Not sure if that is how it is intended to work.